x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable.tsx

```diff
@@ -41,12 +41,31 @@ export class AnomalySwimlaneEmbeddable extends AnomalyDetectionEmbeddable<
   private reload$ = new Subject<void>();
   public readonly type: string = ANOMALY_SWIMLANE_EMBEDDABLE_TYPE;
 
+  // API
+  public viewBy = new BehaviorSubject<string | undefined>(this.getInput().viewBy);
+  public perPage = new BehaviorSubject<number | undefined>(this.getOutput().perPage);
+  public fromPage = new BehaviorSubject<number | undefined>(this.getOutput().fromPage);
+
+  private apiSubscriptions = new Subscription();
+
   constructor(
     initialInput: AnomalySwimlaneEmbeddableInput,
     public services: [CoreStart, MlDependencies, AnomalySwimlaneServices],
     parent?: IContainer
   ) {
     super(initialInput, services[2].anomalyDetectorService, services[1].data.dataViews, parent);
+
+    this.apiSubscriptions.add(
```

Comment:
I think you will want to add `distinctUntilKeyChanged` so that BehaviorSubject is only updated when the key changes, and not everytime the output or input changes. You should just be able to use `embeddableInputToSubject` and `embeddableOutputToSubject` utility methods from src/plugins/embeddable/public/lib/embeddables/compatibility/embeddable_compatibility_utils.ts